app/controllers/main_controller.rb

```diff
@@ -317,6 +316,15 @@ def clear_role_switch_session
     redirect_to action: 'login'
   end
 
+  def check_timeout
+    if !check_warned and check_imminent_expiry
+      render template: "main/timeout_imminent"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.